modules/light-clients/08-wasm/keeper/keeper.go

```diff
@@ -38,7 +38,7 @@ func NewKeeperWithVM(
 	cdc codec.BinaryCodec,
 	key storetypes.StoreKey,
 	authority string,
-	vm *wasmvm.VM,
+	vm types.WasmEngine,
```

Comment:
Are we okay not adding an assertion that the user provides the canonical wasmvm type?

Reply:
I'd be fine without it. Tests that mock this would fail if we had this assertion too no?

Reply:
yea, you'd probably need to avoid using the `NewKeeper` funcs for mocks